### PR TITLE
feat: implement apps and app store communication

### LINKED ISSUE #445: Browsing and install IDS Apps
**Describe the solution you'd like**
- [ ] Adding App Stores to the DSC/CM (same as for brokers now)
- [ ] Implementing an API to search for Data Apps in corresponding App Stores
- [ ] Download a specific App
- [ ] Install it to the Docker environment

Make apps integrable into routes is currently out of scope.
But feel free to add it when there is free time ;-)

### PR DESCRIPTION
**Pull request by vdakker**
closes #445

### CODE REVIEW

**Comment by juliapampus** [comment]
The `DataApp` should be named `App`. In addition, we did not talk about creating an `AppRepresentation`. Open todos are listed at slide 6 in our brainstorming pptx. @vdakker

**Comment by ghost** [comment]
Just a little warning: With the current combination of inbound and outbound model version on DSC main, the AppStore will respond with an "Infomodel Version of incoming Message not supported" RejectionMessage. If compatibility should remain as currently in this branch, the AppStore would need to be updated.

**Comment by juliapampus** [comment]
> Just a little warning: With the current combination of inbound and outbound model version on DSC main, the AppStore will respond with an "Infomodel Version of incoming Message not supported" RejectionMessage. If compatibility should remain as currently in this branch, the AppStore would need to be updated.

I agree with you for the offical release. At least the inbound versions of the AppStore should be updated @DominicReuter-FIT. For local tests, it should be enough to just change the config.json accordingly.

File: src/main/java/io/dataspaceconnector/model/artifact/ArtifactImpl.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 @SQLDelete(sql = "UPDATE artifact SET deleted=true WHERE id=?")
 @Where(clause = "deleted = false")
 @Getter
-@Setter(AccessLevel.PACKAGE)
+@Setter(AccessLevel.PUBLIC)
```
Comment (brianjahnke) [inline src/main/java/io/dataspaceconnector/model/artifact/ArtifactImpl.java]: The artifact's data should not be publicy accessible. If access is needed to the data the ArtifactService should be used.

Comment (vdakker) [reply src/main/java/io/dataspaceconnector/model/artifact/ArtifactImpl.java]: new commit changed the setters accesslevel to package

**Review by juliapampus — APPROVED** [review]

**Pull request merged**